### DISCUSSION TITLE
Disable replica count when autoscale is defined

### DISCUSSION THREAD

**Discussion by RomanDvorsky** [Azure/api-management-self-hosted-gateway#235] [Q&A]
### Proposal

Hi, 

we have added autoscale by keda to our self hosted gateway deployment however we have problems to disable replica count and now we are facing typical problem when our self-hosted APIM is scaled up the new deployment rescale replicas to replica count defined in the helm chart.

### Use-Case

Anyhow disable the replica count so the replicas are controlled by keda.

**Comment by tomkerkhove** [comment]
Can you elaborate on how you are using KEDA to scale it and what your configuration is? Normally replica count should not cause issues as KEDA just uses HPA under the hood :)

_PS: Thanks for using KEDA from a KEDA maintainer 👋_

**Reply by RomanDvorsky** [reply]
Hi, this is the scenario:

1. cool down for scale is set to 1 hour (just to stabilize the scale down window)
2. we had replica count to set to 15 instanced (really defensive)
3. our service has been scaled to 3 instances
4. new deployment scale the apim to 15 instances (due replicacount is set)
5. after 1 hour it was scaled down

if the replicacount is not set there will be no scale up and down I suppose, right?

**Reply by RomanDvorsky** [reply]
here is the graph with number of pod for self hosted gateway - as you can see the spike to 15 instances is new deployment (which we would like to avoid):
![image](https://github.com/Azure/api-management-self-hosted-gateway/assets/36188474/d9846471-2876-4a9f-b76d-4c31f8368031)

**Reply by tomkerkhove** [reply]
What do you mean with "new deployment scale the apim to 15 instances (due replicacount is set)"? Can you elaborate on what you mean with "new deployment"?

**Reply by RomanDvorsky** [reply]
yep when we change any settings for the apim helm chart than the new deployment is rolled out

**Reply by RomanDvorsky** [reply]
actually what we want to achieve is how to write logs to log analytics - so we did change the dapr set up (but this is another [bummer](https://github.com/Azure/api-management-self-hosted-gateway/discussions/238)) -maybe we can set new discussion for this topic

**Reply by tomkerkhove** [reply]
Feel free to send a PR to make replica count optional if you want - Thanks!
